src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1935,7 +1939,11 @@ private static void ReportCtorInitializerCycles(MethodSymbol method, BoundExpres
                 }
             }
 
-            if (constructor is SynthesizedRecordCopyCtor copyCtor)
+            if (constructor is SynthesizedRecordConstructor && constructor.ContainingType.IsStructType())
```

Comment:
>constructor is SynthesizedRecordConstructor && constructor.ContainingType.IsStructType() [](start = 16, length = 88)

This is not blocking, but I would simply change this condition to check if the containing type is a structure or an enum.